Plumbing in support to be a portable web application

loudej opened the pull request:
Uses OWIN server abstractions

izevaka committed:
Moved common API response code into extension methods.

- Added QueryString extension method that tokenizes URL querystring
  parameters.
- Added TryGetAndConvert method that gets a querystring parameter and tries
  to convert it to a specified type
- Created a whole bunch of extension methods to return success and error
  responses.

loudej committed:
Shifting from ASP.NET hosting to all hosts via OWIN

Switching SignalR and WebApi host binding to OWIN adapters
Adding packages for Gate middleware and ASP.NET support
Adds TransitionalGlue.cs to minimize code impact

Conflicts:

	JabbR/packages.config

loudej committed:
Patching in the components that do request work

Use(Sanity) provides early warning of incorrect call patterns
UseHomePage calls a simplistic default.aspx renderer
UseStatic returns filesystem objects for requests
UseSignalRHubs dispatches requests to signalr
RunHttpServer dispatches to WebAPI constrollers

loudej committed:
Updated to fix ninject scope

Adds a scope token object to execution context
NewExecutionContext middleware forces an exec context per request
UseRequestScope middleware adds and removes scope token
Path /Default.aspx is also added as a home page url

davidfowl committed:
Added support for ssl packaging to azure.

davidfowl committed:
Added missing ssl cert binding.

samandmoore commented on JabbR/Startup.cs:
```diff
@@ -0,0 +1,184 @@
+                                        context.CreateCopy(),
+                                        __ => body(write, flush, end, cancellationToken),
+                                        null)),
+                                fault);
```
this is really difficult to read, is there a better way to format this?

loudej replied:
I removed it altogether

that particular middleware was just being defensive - in case the http server didn't provide a new ExecutionContext per http connection. In the end it probably makes more sense to require the http server handlers for firefly and gate to ensure there's a new execution context that "sticks" to each of the calls coming in for a request.

davidfowl committed:
Changed cert name to jabbr.

izevaka committed:
Refactored UriExtensions to be a non-static class.

- Created a QueryString class that wraps up extracting and converting query
  string parameters.

loudej committed:
Refactoring for clarity and adding comments

A few concerns, like setting the ASP.NET current directory, and
having a distinct ExecutionContext per request, have been moved
to their respective host handlers

davidfowl committed:
Detect https for signalr traffic by requesting an image on the site via https.
If it fails, don't bother changing the url. Otherwise use the https url.

davidfowl committed:
Remove detection logic for https since it's blocked because of cors.

davidfowl committed:
Made youtube embed urls use https

davidfowl committed:
Make user voice content provider work with https.

davidfowl committed:
Use https for twitter profile images.

davidfowl committed:
Ah forgot the jsonp call for twitter :).

davidfowl committed:
Added proxying for images and imgur.
- Added very basic proxy handler and made imgur and images.

izevaka committed:
Removed common setup from unit test constructors.

davidfowl committed:
Changed proxy urls from Proxy.ashx to proxy so no implementation is assumed.
- Changed the image content providers appropriately.

izevaka committed:
Cleaned up variable names for consistency sake.

- Changed _QueryStringCollection local variable to queryStringCollection.
- Changed _QueryStringDictionary member variable to _queryStringDictionary.

davidfowl committed:
Removed dead code.

davidfowl committed:
Remove dead code from chat.js that tries to detect https since it can't work.

loudej committed:
Fixing merge conflict between RouteMagic and Owin reference

loudej commented:
This will take some refactoring at other levels before a pull request to jabbr is practical.

Thanks for the feedback though!
